modules/ROOT/pages/batch-code-example.adoc

```diff
@@ -0,0 +1,137 @@
+. Import your downloaded example into Studio. +
+See xref:7.2@studio::import-export-packages.adoc[Importing and Exporting Projects] for more information. +
+Additionally, you can import this very same example from Studio. See xref:7.2@studio::import-project-exchange.adoc[To Import a Mule Project from Exchange] for more details.
+. Open the *mule-artifact.properties* file inside the src/main/resources folder and complete the `sfdc.user`, `sfdc.password`, and `sfdc.securityToken` with your Salesforce's username, password and security token respectively.
+
+== Batch Example Code
+
+The application accepts a CSV file which contains information about the leads you want to insert – company, first name, last name, birthday, email – and uploads them into a Salesforce account, automatically inserting the correct data into each Salesforce field.
+
+.Batch example in the Studio canvas.
+image::batch-code-exmaple-10b3f.png[]
+
+The File connector polls the input folder for new files every ten seconds. When it spots a new file, it reads it and passes the content to the Transform Message processor. +
+This processor converts the format of the data from CSV to a collection, and it automatically maps the input fields from the CSV file – FirstName, LastName, etc. – to output fields that Salesforce uses in a collection.
+
+.XML Sample of the application setting up the data
+[source,xml,linenums]
+----
+<flow name="import-leads-into-salesforceFlow" >
+
+  <scheduler>
+    <scheduling-strategy>
+      <fixed-frequency frequency="${scheduler.frequency}" startDelay="${scheduler.startDelay}" />
+    </scheduling-strategy>
+  </scheduler>
+
+  <file:read config-ref="File_Config" path="leads.csv" outputMimeType="text/csv">
+    <ee:repeatable-file-store-stream />
+  </file:read>
+
+  <ee:transform doc:name="Transform CSV to Maps">
+    <ee:message>
+      <ee:set-payload>
+        <![CDATA[%dw 2.0
+        output application/java
+        ---
+        payload map {
+          Company    : $.company,
+          Email      : $.email,
+          FirstName  : $.firstName,
+          LastName   : $.lastName
+        }]]>
+      </ee:set-payload>
+    </ee:message>
+  </ee:transform>
+----
+
+After all leads in the file are converted to a collection of Salesforce-friendly data, the application uses a batch job scope to go through each lead in the CSV file and insert it into your Salesforce account.
+It starts with the batch step `LeadExistsStep`, where a Salesforce Connector queries for existing leads. Note that it uses a target variable called "leadNotFound" to store the CSV lead values that were not found in your Salesforce account.
+
+[source,xml,linenums]
+----
+<batch:job jobName="CreateLeadsBatch" maxFailedRecords="1000">
+  <batch:process-records >
+    <batch:step name="LeadExistsStep">
+      <salesforce:query config-ref="Salesforce_Sfdc_config" target="leadNotFound" targetValue="#[isEmpty(payload)]">
```

Comment:
this is a very bad example. Doing this on a per record basis will consume a large amount of the SFDC API quota. Need to validate with a salesforce expert if there's a better way of doing this or find a better example.